### ISSUE TITLE
Real Debrid now limit API calls

### ISSUE THREAD

**Issue by minicoz** [Tikipeter/tikipeter.github.io#54]
https://i.imgur.com/KNEMWWB.jpeg

Seems like Real Debrid has just implemented some API limits. 

Some users on Reddit have reported Fen Light not working anymore. 

I love this app, just informing you Tiki. I don't have a solution as of now.

**Comment by minicoz** [comment]
https://x.com/RealDebrid/status/1811721271677181980

**Comment by Tikipeter** [comment]
>Some users on Reddit have reported Fen Light not working anymore.

Fen/FenLight will work fine once RD gets everything back on it's feet.

>I don't have a solution as of now.

I wouldn't worry. There is absolutely nothing that I need to do to have Fen/FenLight continue working with RD.

It's important to not believe things you read on the internet, particularly a swamp like Reddit where EVERYONE is an expert on EVERYTHING. 

```
We are having some difficulties with our database, we are working on it.
```

This is the only thing RD have written that is relevant at all to Fen/FenLight (or most any Kodi addon really) during this time.

>Seems like Real Debrid has just implemented some API limits.

That's fine.

If the limit is 250 calls a minute, then at the point when Fen/FenLight is most calling RD (after scraping and during resolving) then Fen/FenLight would be at least 240 calls a minute under that figure.

There's 1 call to RD to get the cached status of the torrents found in the scrape, and then there are 5 calls to RD related to the resolving process. So normally 6 total calls in the span of a minute to check cache status and play a found file from RD.

Of course, if you're not scraping results or resolving a link for playback, then Fen/FenLight is making 0 calls to RD's API. If you watch 2 movies in the evening, that would be a total of 12 calls to RD's API over the course of the 3 hours of watching the 2 videos.

Kodi addons are not the problem here. There are scripts/programs floating around whose name I can't remember that is adding every video torrent it can find (that means a LOT) to a user's RD Cloud, and then continually re-does this so that the torrents in the user's cloud remains active.

It is a flagrant abuse of RD's API, and I fully support them in trying to mitigate the damage these scripts/programs, and the people using them, cause to their service.

Enough's enough with these assholes. Kodi addons have been living harmoniously with all the API's we use for many, many years.

Long story short.... there's nothing to worry about.

**Comment by Tikipeter** [comment]
>It’s clear the current versions of these addons are not working within the new limits put in place. Updates required on the addons side unless RD increase the limits again once resources are under control.

That's the sort of piffle addons4kodi will give you. Could not be more wrong if he/she tried.

**Comment by Tikipeter** [comment]
![8wp59v](https://github.com/user-attachments/assets/8699490e-b6af-4408-b4a3-ec7c342dcb6e)

**Comment by hcgiub001** [comment]
It's kind of crazy users think addons are to blame, I don't get why a Dev who works for free creating free addons will risk abusing a server. what possible reason would they do that? also all popular addons are open source, you don't think RD is aware of the addons and checked what the code is doing? They would have put a stop to this if they saw any blatant misuse of the servers and circumvented addons from doing so. 
people lack critical thinking. 

> > It’s clear the current versions of these addons are not working within the new limits put in place. Updates required on the addons side unless RD increase the limits again once resources are under control.
> 
> That's the sort of piffle addons4kodi will give you. Could not be more wrong if he/she tried.
> 
> This is fun. I'll add some more nonsense that's been written. Just for my own amusement.
> 
> > RD has made changes to their API which have broken some if not all add-ons depending on the way they use the API. The add-ons will need to be updated before they start working again.
> 
> > Umbrella. Asgard. Fen. seren. Are all affected of the new rules that real-debrid put in place as far as i can tell atm
> 
> > We are at the mercy of the guys that maintain the addon's to update to the new API and push out these updates.......Come on Fen/CoCo team!
> 
> > Looks like they avoiding resource meltdown and have increased limits again.
> 
> > Anyone else having issues?
> 
> > You need to reauthorize your RD (even if it's authorized now, revoke it and redo) so it'll get a new API key in whatever add-on you're using.
> 
> > Umbrella. Asgard. Fen. seren. Are all affected of the new rules that real-debrid put in place as far as i can tell atm
> 
> The Question:
> 
> > Anyone know which apps are doing this so we can be sure to stay away?
> 
> The Answer:
> 
> > POV is working for me
> 
> > Umbrella. Asgard. Fen. seren. Are all affected of the new rules that real-debrid put in place as far as i can tell atm
> 
> Etc, etc....

It's kind of crazy users think addons are to blame, I don't get why a Dev who works for free creating free addons will risk abusing a server. what possible reason would they do that? also all popular addons are open source, you don't think RD is aware of the addons and checked what the code is doing? They would have put a stop to this if they saw any blatant misuse of the servers and circumvented addons from doing so. 
people lack critical thinking.

**Comment by hcgiub001** [comment]
LOL I thought you was actually talking about Managers at Debrid. I've never heard of the app Debrid Media Manager lol
